hoomd/mpcd/DryingDropletStreaming.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2009-2019 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: mphoward
+
+/*!
+ * \file mpcd/DryingDropletStreaming.h
+ * \brief Declaration of mpcd::DryingDropletStreaming
+ */
+
+#ifndef MPCD_DRYING_DROPLET_STREAMING_H_
+#define MPCD_DRYING_DROPLET_STREAMING_H_
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "ConfinedStreamingMethod.h"
+#include "hoomd/extern/pybind/include/pybind11/pybind11.h"
+
+namespace mpcd
+{
+
+//! MPCD confined streaming method
+/*!
+ * This method implements the base version of ballistic propagation of MPCD
+ * particles in Spherical geometry.
+ *
+ * \tparam Geometry The confining geometry (SphereGeometry).
+ *
+ * The integration scheme is essentially Verlet with specular reflections. The particle is streamed forward over
+ * the time interval. If it moves outside the Geometry, it is placed back on the boundary and its velocity is
+ * updated according to the boundary conditions. Streaming then continues until the timestep is completed.
+ *
+ * To facilitate this, every Geometry must supply three methods:
+ *  1. detectCollision(): Determines when and where a collision occurs. If one does, this method moves the
+ *                        particle back, reflects its velocity, and gives the time still remaining to integrate.
+ *  2. isOutside(): Determines whether a particles lies outside the Geometry.
+ *  3. validateBox(): Checks whether the global simulation box is consistent with the streaming geometry.
+ *
+ */
+template<class Geometry>
+class PYBIND11_EXPORT DryingDropletStreaming : public mpcd::ConfinedStreamingMethod
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param geom Streaming geometry
+         * \param R is the radius of sphere intially at very first timestep
+         */
+        DryingDropletStreaming(std::shared_ptr<mpcd::SystemData> sysdata,
+                                unsigned int cur_timestep,
+                                unsigned int period,
+                                int phase,
+                                std::shared_ptr<const Geometry> geom, Scalar R)
+        : mpcd::ConfinedStreamingMethod(sysdata, cur_timestep, period, phase),
```

Comment:
I think you will want to take only `R` and make it a `Variant`. You can initialize the geometry as a null pointer for now, then you can initialize it properly later.
```suggestion
                                Variant R)
        : mpcd::ConfinedStreamingMethod<mpcd::SphericalGeometry>(
            sysdata, cur_timestep, period, phase, std::shared_ptr<const mpcd::SphericalGeometry>()),
        m_R(R)
```